src/effects/effectslot.cpp

```diff
@@ -268,3 +272,62 @@ void EffectSlot::slotEffectMetaParameter(double v, bool force) {
         pParameterSlot->onEffectMetaParameterChanged(v, force);
     }
 }
+
+QDomElement EffectSlot::toXml(QDomDocument* doc) const {
+    QDomElement effectElement = doc->createElement(EffectXml::Effect);
+    if (m_pEffect == nullptr) {
+        return effectElement;
+    }
+
+    QDomElement metaKnobElement = doc->createElement(EffectXml::EffectMetaParameter);
+    XmlParse::addElement(*doc, effectElement,
+                         EffectXml::EffectMetaParameter,
+                         QString::number(m_pControlMetaParameter->get()));
+    EffectManifest manifest = m_pEffect->getManifest();
+    XmlParse::addElement(*doc, effectElement,
+                         EffectXml::EffectId, manifest.id());
+    XmlParse::addElement(*doc, effectElement,
+                         EffectXml::EffectVersion, manifest.version());
+
+    QDomElement parametersElement = doc->createElement(EffectXml::ParametersRoot);
+
+    for (const auto& pParameter : m_parametersById) {
```

Comment:
Since m_parametersById is only used in serializing/deserializing, I wonder if it makes sense to build it on the fly instead of storing at the class level (which requires keeping it in sync as new effects are loaded). 

Also, why a QMap? did you want to the items to be sorted by ID in the XML (instead of being in the parameter order in the manifest).